lib/matplotlib/backend_bases.py

```diff
@@ -138,6 +145,35 @@ def get_registered_canvas_class(format):
     return backend_class
 
 
+class MainLoopBase(object):
```

Comment:
mpl 3.0, really want to hold the line on 2.0 is style changes only.

I am sold on not using an ABC here. 

I seem to recall the same thing about the pdf backend last time I looked at this but do not remember the details.